server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1702,4 +1711,7 @@ public boolean allPendingDanglingIndicesWritten() {
         return timestampFieldMapperService.getTimestampFieldType(index);
     }
 
+    public Function<IndexMetadata, TimeSeriesIdGenerator> getTimeSeriesGeneratorLookup() {
```

Comment:
I use the `Function` here so its easier to test the bulk action without standing up a bunch of stuff.